src/HealthGPS/energy_balance_model.cpp

```diff
@@ -78,131 +82,169 @@ void EnergyBalanceModel::update_risk_factors(RuntimeContext &context) {
+    double RMR = compute_RMR(BW_0, H, person.age, person.gender);
 
     double delta_0 = ((1.0 - beta_TEF) * PAL - 1.0) * RMR / BW_0;
 
-    // Update thermic effect of food and adaptive thermogenesis.
+    // Compute thermic effect of food and adaptive thermogenesis.
     double TEF = beta_TEF * EI - EI_0;
     double AT = beta_AT * EI - EI_0;
 
-    // TODO: Update intercept value.
+    // TODO: Compute intercept value.
     const double K = 0.0; // TODO: Intercept value.
 
     // Energy partitioning.
     const double C = 10.4 * rho_L / rho_F;
     const double p = C / (C + F_0);
 
+    double x = p * eta_L / rho_L + (1.0 - p) * eta_F / rho_F;
+
     // First equation ax + by = e.
     double a1 = p * rho_F;
     double b1 = -(1.0 - p) * rho_L;
     double c1 = p * rho_F * F_0 - (1 - p) * rho_L * L_0;
 
-    // Check no baseline adjustment in trial run.
-    if (!final_run)
-        final_shift = 0.0;
-
     // Second equation cx + dy = f.
     double a2 = gamma_F + delta_0;
     double b2 = gamma_L + delta_0;
-    double c2 = EI - final_shift - K - TEF - AT - delta_0 * (G + W + ECF);
+    double c2 = EI - shift - K - TEF - AT - delta_0 * (G + W + ECF);
 
-    // Update body fat and lean tissue steady state.
+    // Compute body fat and lean tissue steady state.
     double steady_F = -(b1 * c2 - b2 * c1) / (a1 * b2 - a2 * b1);
     double steady_L = -(c1 * a2 - c2 * a1) / (a1 * b2 - a2 * b1);
 
-    double x = p * eta_L / rho_L + (1.0 - p) * eta_F / rho_F;
+    // Compute time constant.
     double tau = rho_L * rho_F * (1.0 + x) /
                  ((gamma_F + delta_0) * (1.0 - p) * rho_L + (gamma_L + delta_0) * p * rho_F);
 
-    // Update body fat and lean tissue.
-    double F = steady_F + (F_0 - steady_F) * exp(-365.0 / tau);
-    double L = steady_L + (L_0 - steady_L) * exp(-365.0 / tau);
+    // Compute body fat and lean tissue.
+    double F = steady_F - (steady_F - F_0) * exp(-365.0 / tau);
+    double L = steady_L - (steady_L - L_0) * exp(-365.0 / tau);
+
+    // Compute baseline adjustment coefficient.
+    double adjust = -(a1 - b1) * (1.0 - exp(-365.0 / tau)) / (a1 * b2 - a2 * b1);
 
-    // Update body weight.
+    // Compute body weight.
     double BW = F + L + G + W + ECF;
 
+    // Compute energy expenditure.
     double delta_BW = delta_0 * BW;
+    double EE = (shift + K + gamma_F * F + gamma_L * L + delta_BW + TEF + AT + EI * x) / (1.0 + x);
+
+    // New simulated person state.
+    SimulatePersonState state{.H = H,
+                              .BW = BW,
+                              .PAL = PAL,
+                              .RMR = RMR,
+                              .F = F,
+                              .L = L,
+                              .ECF = ECF,
+                              .G = G,
+                              .W = W,
+                              .EE = EE,
+                              .EI = EI,
+                              .adjust = adjust};
+
+    return state;
```

Comment:
Don't commit this suggestion as `clang-format` will complain, but I'd do this in one line like so:

```suggestion
    // New simulated person state.
    return SimulatePersonState{.H = H,
                              .BW = BW,
                              .PAL = PAL,
                              .RMR = RMR,
                              .F = F,
                              .L = L,
                              .ECF = ECF,
                              .G = G,
                              .W = W,
                              .EE = EE,
                              .EI = EI,
                              .adjust = adjust};
```

Reply:
It's quite a cheap op anyway. I'll just leave it for now.